corehq/motech/fhir/tests/test_api_endpoints.py

```diff
@@ -0,0 +1,290 @@
+"""
+These tests set goals for the functionality of CommCare's FHIR API.
+
+Examples are based on HAPI FHIR. You can explore requests and responses
+using their `online test server`_.
+
+
+.. _online test server: https://hapi.fhir.org/resource?serverId=home_r4&pretty=false&_summary=&resource=Patient#
+
+"""
+from unittest import skip
+from uuid import uuid4
+
+from django.test import TestCase
+
+import requests
+
+from casexml.apps.case.mock import CaseBlock
+
+from corehq.apps.accounting.models import SoftwarePlanEdition
+from corehq.apps.accounting.tests.utils import DomainSubscriptionMixin
+from corehq.apps.accounting.utils import clear_plan_version_cache
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.tzmigration.timezonemigration import json_diff
+from corehq.apps.users.models import WebUser
+from corehq.motech.utils import get_endpoint_url
+
+DOMAIN = 'fhir-drill'
+FHIR_VERSION = 'R4'  # We do not need to support older versions (e.g. DSTU2)
+# MVP: API is domain-specific, like existing API.
+BASE_URL = f'localhost:8080/a/{DOMAIN}/api/fhir/{FHIR_VERSION}/'
+API_USERNAME = f'admin@{DOMAIN}.commcarehq.org'
+API_PASSWORD = 'Passw0rd!'
+FOO_CASE_ID = uuid4().hex
+BAR_CASE_ID = uuid4().hex
+BAZ_CASE_ID = uuid4().hex
+
+
+@skip('Covered functionality not yet implemented')
+class PatientEndpointTests(TestCase, DomainSubscriptionMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(DOMAIN)
+        # Assumes FHIR API needs Pro plan
+        cls.setup_subscription(DOMAIN, SoftwarePlanEdition.PRO)
+        cls.user = WebUser.create(DOMAIN, API_USERNAME, API_PASSWORD,
+                                  created_by=None, created_via=None)
+        create_person_cases(owner_id=cls.user.user_id)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete(deleted_by=None)
+        cls.teardown_subscriptions()
+        cls.domain_obj.delete()
+        clear_plan_version_cache()
+        super().tearDownClass()
+
+    def test_search(self):
+        # Tests filtering by a non-indexed case property
+        url = get_endpoint_url(BASE_URL, '/Patient?birthdate=1990-01-01')
+        response = requests.get(url, auth=(API_USERNAME, API_PASSWORD))
+
+        # Some things will be different between the expected search
+        # result and the actual response. Use `json_diff()` to confirm
+        # expected differences, and that everything else is the same.
+        diffs = sorted([diff.path for diff in json_diff(response.json(),
+                                                        get_search_result())])
+        self.assertEqual(diffs, ['id', 'meta.lastUpdated'])
```

Comment:
Can you add why these two would be expected to be different?
I guess
1. `id` is generated new each time and hence would be different
2. `meta.lastUpdated` would be the time any of the case in that search result was updated? that would be a hard stat to get for a quick search. But also, its different here because the test has a hard coded value in the code?

Reply:
Exactly right on both counts. Thanks, I'll add to the comment.

Reply:
6af81d1